src/backend/src/services/projects.services.ts

```diff
@@ -843,6 +843,38 @@ export default class ProjectsService {
     return newMaterialType;
   }
 
+  /**
+   * Deletes an assembly type
+   * @param assemblyId the name of the assembly
+   * @param submitter the user who is deleting the assembly type
+   * @throws if the user is not an admin/head, the assembly does not exist, or has already been deleted
+   * @returns
+   */
+  static async deleteAssembly(assemblyId: string, submitter: User): Promise<Assembly> {
+    if (!isAdmin(submitter.role) || !isHead(submitter.role))
+      throw new AccessDeniedException('Only an Admin or a head can delete an Assembly');
+
+    const assembly = await prisma.assembly.findUnique({
+      where: {
+        assemblyId
```

Comment:
this was "name" before, but you were inputting the assemblyId so it wasn't working